daemon/socket/proxy.go

```diff
@@ -48,10 +48,10 @@ type AuthProxy struct {
 
 // NewAuthProxy returns a new AuthProxy. It will return an error if creation
 // of the domain socket fails, or the upstream registry URL is misconfigured.
-func NewAuthProxy(c *config.Config, sess session.Session, db *db.DB,
```

Comment:
Can we get a doc string about the different the socket is configured (read/write user/group)?